File: app/src/main/java/com/example/marvel/view/DetailCharacters.java
package com.example.marvel.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;
import com.example.marvel.R;
import com.example.marvel.model.characters.ResultsCharacters;
import com.example.marvel.model.comics.ResultsComics;
import com.squareup.picasso.Picasso;

public class DetailCharacters extends AppCompatActivity {

    private ImageView imageCharacter;
    private ImageView imageBackground;
    private TextView title, descripiton;
    private Button favorite;
    ResultsCharacters.Data.Characters character;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.characters_detail);

        Window window = getWindow();
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        window.setStatusBarColor(getColor(R.color.detailColor));
        getWindow().setNavigationBarColor(ContextCompat.getColor(this, R.color.detailColor));


        imageCharacter = findViewById(R.id.img_characters);
        imageBackground = findViewById(R.id. img_characters_two);
        title = findViewById(R.id.title_characters);
        descripiton = findViewById(R.id.description_characters);
        favorite = findViewById(R.id.btn_character);

        if(getIntent() != null){
            ResultsCharacters.Data.Characters characters = getIntent().getExtras().getParcelable("Hero");
            if(characters.getDescription().isEmpty()) {
                descripiton.setText(R.string.descri_off);
            } else {
                descripiton.setText(characters.getDescription());
            }
            title.setText(characters.getName());
            Picasso.get().load(characters.getThumbnail().getPath() + ".jpg").error(R.drawable.logo_aplicativo).into(imageBackground);
            Picasso.get().load(characters.getThumbnail().getPath() + ".jpg").error(R.drawable.logo_aplicativo).into(imageCharacter);
        }else {
            Toast.makeText(this, R.string.problem_server, Toast.LENGTH_LONG).show();
        }
        imageCharacter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), ImageCharactersActivity.class);
                Bundle bundle = new Bundle();
                bundle.putParcelable("character", character);
                startActivity(intent.putExtras(bundle));
            }
        });
    }
}

File: app/src/main/java/com/example/marvel/view/DetailCharacters.kt
package com.example.marvel.view

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.lifecycle.ViewModelProviders
import com.example.marvel.R
import com.example.marvel.model.characters.CharacterRoom
import com.example.marvel.model.characters.ResultsCharacters
import com.example.marvel.model.characters.ResultsCharacters.Data.Characters
import com.example.marvel.viewmodel.ViewModelMarvel
import com.example.marvel.viewmodel.ViewModelRoom
import com.squareup.picasso.Picasso

class DetailCharacters : AppCompatActivity() {

    private lateinit var imageCharacter: ImageView
    private lateinit var imageBackground: ImageView
    private lateinit var title: TextView
    private lateinit var descripiton: TextView
    private lateinit var favorite: Button
    private lateinit var character: Characters
    private val activity = this
    private val viewModelRoom : ViewModelRoom by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.characters_detail)

        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
        window.statusBarColor = getColor(R.color.detailColor)
        window.navigationBarColor = ContextCompat.getColor(this, R.color.detailColor)

        imageCharacter = findViewById(R.id.img_characters)
        imageBackground = findViewById(R.id.img_characters_two)
        title = findViewById(R.id.title_characters)
        descripiton = findViewById(R.id.description_characters)
        favorite = findViewById(R.id.btn_character)

        if (intent != null) {
            character = intent.extras?.getParcelable("Hero") ?: Characters()
            if (character.description.isNullOrEmpty()) {
                descripiton.setText(R.string.descri_off)
            } else {
                descripiton.text = character.description
            }
            title.text = character.name
            Picasso.get().load(character.thumbnail!!.path + ".jpg").error(R.drawable.logo_aplicativo).into(imageBackground)
            Picasso.get().load(character.thumbnail!!.path + ".jpg").error(R.drawable.logo_aplicativo).into(imageCharacter)
        } else {
            Toast.makeText(this, R.string.problem_server, Toast.LENGTH_LONG).show()
        }
        imageCharacter.setOnClickListener {
            val bundle = Bundle()
            bundle.putParcelable("Hero", character)
            startActivity(Intent(this, ImageCharactersActivity::class.java).putExtras(bundle))
        }

        favorite.setOnClickListener {
            val characterRoom = CharacterRoom(character.description, character.id, character.name, character.thumbnail?.path + ".jpg")
            viewModelRoom.insertCharacterRoom(activity, characterRoom)
            Toast.makeText(activity, "Add aos favoritos", Toast.LENGTH_LONG).show()
        }
    }
}